app/jobs/hyrax/grant_read_job.rb

```diff
@@ -5,10 +5,19 @@ class GrantReadJob < ApplicationJob
 
     # @param [String] file_set_id - the identifier of the object to grant access to
     # @param [String] user_key - the user to add
-    def perform(file_set_id, user_key)
-      file_set = ::FileSet.find(file_set_id)
-      file_set.read_users += [user_key]
-      file_set.save!
+    # @param [Boolean] use_valkyrie - use valkyrie resources for this operation?
+    def perform(file_set_id, user_key, use_valkyrie: Hyrax.config.use_valkyrie?)
+      if use_valkyrie
+        file_set_id = Valkyrie::ID.new(file_set_id)
```

Comment:
Looks good. There's a comment at https://github.com/samvera/hyrax/pull/4118 that shows off a little grant DSL that may be appropriate for this case.

As I said there, I'll leave it up to y'all whether that DSL is useful for this case. 

See also: https://github.com/samvera/hyrax/wiki/Hyrax-Valkyrie-Usage-Guide#grantrevoke

Reply:
This comment? https://github.com/samvera/hyrax/pull/4117#discussion_r337204897